File: cron/whine.pl
#!/usr/bin/perl -l
use strict;
use DBI;
use Net::Ping;

use MIME::Lite;

my $host = "http://tracker.villagesoup.com";
#use Switch;
use LWP::Simple;
my $dbh = DBI->connect('dbi:mysql:trackerVS','root','') or die "Connection Error: $DBI::errstr\n";

my $sthTickets = $dbh->prepare("select * from ticket where ownerId=? and statusId not in (4,6) and dueDate<=? and dueDate !='0000-00-00'");
my $sthUsers = $dbh->prepare("select * from users u inner join  userToGroup utg  on u.userId=utg.userId where utg.userGroupId=2");
my $sthStatus = $dbh->prepare("select * from status where statusId = ?");
my $sthUser = $dbh->prepare("select * from users where userId = ?");
my $sthQueue = $dbh->prepare("select * from queue where queueId=?");
$sthUsers->execute();
my ($sec,$min,$hour,$mday,$mon,$year,$wday,$yday,$isdst) = localtime(time+(3 * 86400));
my $today = sprintf "%.4d-%.2d-%.2d", $year+1900, $mon+1, $mday;
print $today."\n";
my $user;
while ($user = $sthUsers->fetchrow_hashref)
{

	$sthTickets->execute($user->{userId},$today);
	if ($sthTickets->rows > 0)
	{
		print $user->{fullName}."\n";

   	    my $msg = "  <table id='myTickets' class='tablesorter' width='100%'>
                      <tr>
                        <th>Ticket</th>
                        <th>Subject</th>
                        <th>Status</th>
                        <th>Requestor</th>
                        <th>Queue</th>
                        <th>Last Updated</th>
                      </tr>\n";

		my $ticket;
		while ($ticket = $sthTickets->fetchrow_hashref)
		{
			$sthStatus->execute($ticket->{statusId});
			my $status = $sthStatus->fetchrow_hashref;
			$sthUser->execute($ticket->{requestorId});
			my $requestor = $sthUser->fetchrow_hashref;
			$sthQueue->execute($ticket->{queueId});
			my $queue = $sthQueue->fetchrow_hashref;
			print $ticket->{ticketId}."\n";
			$msg = $msg."<tr  class='mritem'>
                          <td>
                            <a href='$host/ticketEdit/$ticket->{ticketId}'>$ticket->{ticketId}</a>
                          </td>
                          <td>$ticket->{subject}</td>
                          <td>$status->{name}</td>
                          <td>$requestor->{fullName}</td>
                          <td>$queue->{name}</td>
                          <td>$ticket->{lastUpdated}</td>
                         </tr>\n";
		}
		$msg = $msg."</table>";
		my $from = "systems\@villagesoup.com";
		my $subject = "Priority Tickets";
		print $msg."\n";
		    my $to = $user->{'email'};
    my $msg = MIME::Lite->new(
                 From     => $from,
                 To       => $to,
                 #Cc       => $cc,
                 Subject  => $subject,
                 Data     => $msg
                 );

    $msg->attr("content-type" => "text/html");
    $msg->send;

	}

}

File: public_html/process/crons/whine.php
<?php
//
//  Tracker - Version 1.3.0
//
//    Copyright 2012 RaywareSoftware - Raymond St. Onge
//
//  Licensed under the Apache License, Version 2.0 (the "License");
//  you may not use this file except in compliance with the License.
//  You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
//  Unless required by applicable law or agreed to in writing, software
//  distributed under the License is distributed on an "AS IS" BASIS,
//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
//  See the License for the specific language governing permissions and
//  limitations under the License.
//
?>
<?php
include_once "globals.php";

include_once "tracker/user.php";
include_once "tracker/ticket.php";
include_once "tracker/status.php";
include_once "tracker/queue.php";
$host = "http://tracker.villagesoup.com";
$user = new User();

$ok = $user->Get("active = 1");
while ($ok)
{
  $param = "statusId not in (4,6)";
  $param = AddEscapedParam($param,"ownerId",$user->userId);
  $param = $param." and dueDate<='".$today."' and dueDate !='0000-00-00'";
  $ticket = new Ticket();
  $hasTickets = $ticket->Get($param);
  $msg = "  <table id='myTickets' class='tablesorter' width='100%'>
                <tr>
                  <th>Ticket</th>
                  <th>Subject</th>
                  <th>Status</th>
                  <th>Requestor</th>
                  <th>Queue</th>
                  <th>Last Updated</th>
                </tr>\n";
  $hasOverDueTickets = 0;
  while ($hasTickets)
  {
    $status = new Status($ticket->statusId);
    $queue = new Queue($ticket->queueId);
    $requestor = new User($ticket->requestorId);
    $msg = $msg."<tr  class='mritem'>
                        <td>
                          <a href='$host/ticketEdit/$ticket->ticketId'>$ticket->ticketId</a>
                        </td>
                        <td>$ticket->subject</td>
                        <td>$status->name</td>
                        <td>$requestor->fullName</td>
                        <td>$queue->name</td>
                        <td>$ticket->lastUpdated</td>
                       </tr>\n";
    $hasOverDueTickets = 1;
    $hasTickets = $ticket->Next();
  }
  $msg = $msg."</table>";
  if ($hasOverDueTickets)
  {
    echo $msg;
    $headers = 'From: '.$sendersEmail. "\r\n";;
    $to = $user->email;
    $subject = "Testing: Priority Tickets";
    $headers .= "MIME-Version: 1.0\r\n";
    $headers .= "Content-Type: text/html; charset=UTF-8\r\n";
    if (mail($to,$subject,$msg,$headers))
	  {
		  DebugText("mail send successfully to ".$to);
	  }
	  else
	  {
		 DebugText("mail send failed to ".$to);
	  }
  }
  $ok = $user->Next();
}
?>
